### ISSUE TITLE
confusion over responses and unpack_arf/rmf

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1061]
More a note-to-self at the moment. I'm not sure if this can/should be cleaned up or just documented explicitly.

In playing around with PHA responses I've found that the DataPHA.get_responses routine returns different data to unpack_arf (DataARF vs ARF1D). Thanks to the design of ARF1D it isn't a "big" issue in many cases (as ARF1D should just fall-through to the underlying ARF for attribute access, this is unchecked but `__getattr__ has logic in it that suggests this is so). However, it did just catch me out:

```
In [4]: pha.get_response()
Out[4]:
(<DataARF data set instance 'sherpa-test-data/sherpatest/3c273.arf'>,
 <DataRMF data set instance 'sherpa-test-data/sherpatest/3c273.rmf'>)

In [5]: arf, rmf = pha.get_response()

In [6]: help(arf)


In [7]: help(rmf)


In [8]: arf.__class__.__name__
Out[8]: 'DataARF'

In [9]: rmf.__class__.__name__
Out[9]: 'DataRMF'

In [10]: pha.__class__.__name__
Out[10]: 'DataPHA'

In [11]: arf = unpack_arf('sherpa-test-data/sherpatest/3c273.arf')

In [12]: arf.__class__.__name__
Out[12]: 'ARF1D'
```

So `unpack_arf` returns `ARF1D` but the response is `DataARF`.

I was expecting `unpack_arf` to return `DataARF`.

This plays into an underlying question I have over the large number of sherpa.astro.instrument classes and why they are all needed.

**Comment by hamogu** [comment]
I've come across a similar question in the session classes (#796) mad @DougBurke sounds as if the unexplained duplication of functionality might be more common. Maybe @olaurino can give some background or point to an early design experiment that explains the original intent?

**Comment by hamogu** [comment]
When I say "similar" I mean "I don't understand why extra classes are needed or why code is duplicated" in cases that might have nothing to do with the ARF (which is the point of this issue).

**Comment by DougBurke** [comment]
The `ARF1D` is doing something different to the `DataARF` class, as it acts to "add a response to a model expression" when called. So there is some reason for different functionality, but there are a number of similar classes in `sherpa.astro.instrument` and it's hard to follow the intended logic. In this particular question I'm (originally) focused on `unpack_arf`, I guess, and why it doesn't return a `DataARF`.

We could also move the response-generating functionality into `DataARF` and co, but I think that combines to many concerns.

**Comment by hamogu** [comment]
Looking closer at the `ARF1D` there is only a single method that does something: `__call__` (other than passing through to another object). So, this could be a function and not an object (modulo the fact that is need to have a reference to the PHA when called).
Not counting tests, comment strings, and the definition of `ARF1D` itself, the string `ARF1D` appears only three times in the codebase, suggesting that this would not be to radical of a change.

**Comment by DougBurke** [comment]
I'm somewhat reluctant to make "single" changes like this rather than have a better understanding of the whole module and change it all in one go (if we decide to change). For instance there's a `RMF1D` case and a couple of `xxxResponse1D` classes too. We'd need to go through the design docs (the UDF elements) to check.

**Comment by hamogu** [comment]
I'm not opening a PR for "just that single change". I'm just using this issue to collect my observations. I think this is a case where it is actually useful to work the other way around: We have a stable and working, but somewhat convoluted implementation. I'm considering (as in I started playing with it last night for a bit) combining classes in the instrument module and suggesting an alternative implementation. If that works, then we need to go back and compare to the design docs. @wmclaugh recently pointed out that community driven projects are often not as good in having proper design docs, on the other hand, they can be good in "implementing what works". My suspicion is that the original design docs are somewhat overdesigned and lead to a structure that allows for all sorts of further extensions that in practice won't happen and is not needed. In that case, some of that extra complexity might be removed eventually. Anyway, just thinking out loud at this point.
